File: server/api/internal/usecase/interfaces/workerconfig.go
package interfaces

import (
	"context"

	"github.com/reearth/reearth-flow/api/pkg/workerconfig"
)

type WorkerConfig interface {
	FindByWorkspace(context.Context, workerconfig.WorkspaceID) (*workerconfig.WorkerConfig, error)
	GetDefaults(context.Context) (*workerconfig.WorkerConfig, error)
	Update(context.Context, UpdateWorkerConfigParam) (*workerconfig.WorkerConfig, error)
	Reset(context.Context, ResetWorkerConfigParam) (*workerconfig.WorkerConfig, error)
}

type UpdateWorkerConfigParam struct {
	WorkspaceID           workerconfig.WorkspaceID
	BootDiskSizeGB        *int
	BootDiskType          *string
	ChannelBufferSize     *int
	ComputeCpuMilli       *int
	ComputeMemoryMib      *int
	FeatureFlushThreshold *int
	ImageURL              *string
	MachineType           *string
	MaxConcurrency        *int
}

type ResetWorkerConfigParam struct {
	WorkspaceID workerconfig.WorkspaceID
	Fields      []string
}

File: ui/src/types/workerConfig.ts
import { ApiResponse } from "./api";

export type WorkerConfig = {
  id: string;
  machineType?: string;
  computeCpuMilli?: number;
  computeMemoryMib?: number;
  bootDiskSizeGB?: number;
  taskCount?: number;
  maxConcurrency?: number;
  threadPoolSize?: number;
  channelBufferSize?: number;
  featureFlushThreshold?: number;
  nodeStatusPropagationDelayMilli?: number;
  createdAt: string;
  updatedAt: string;
};

export type WorkerConfigMutation = {
  config?: WorkerConfig;
} & ApiResponse;

export type GetWorkerConfig = {
  workerConfig?: WorkerConfig;
  isLoading: boolean;
} & ApiResponse;

export type DeleteWorkerConfig = {
  id: string;
} & ApiResponse;
